frontend/src/components/worksheets/NewRun/NewRun.jsx

```diff
@@ -0,0 +1,475 @@
+                                    placeholder="1"
+                                />
+                            </Grid>
+                            <Grid item xs={6}>
+                                <ConfigLabel
+                                    label="GPUs"
+                                    tooltip="Number of GPUs allocated for this run."
+                                />
+                                <ConfigTextInput
+                                    value={this.state.gpu}
+                                    onValueChange={(value) => this.setState({ gpu: value })}
+                                    placeholder="1"
+                                />
+                            </Grid>
+                            <Grid item xs={12}>
+                                <ConfigLabel
+                                    label="Docker Image"
+                                    tooltip="Tag or digest of Docker image to serve as the
+                                    virtual run environment."
+                                />
+                                <ConfigTextInput
+                                    value={this.state.docker}
+                                    onValueChange={(value) => this.setState({ docker: value })}
+                                    placeholder="codalab/default-cpu:latest"
+                                />
+                            </Grid>
+                            <Grid item xs={12}>
+                                <ConfigSwitchInput
+                                    value={this.state.networkAccess}
+                                    onValueChange={(value) => this.setState({ networkAccess: value })}
+                                />
+                                <ConfigLabel
+                                    label="Network Access"
+                                    tooltip="Whether the bundle can open any external
+                                    network ports."
+                                    inline
+                                />
+                            </Grid>
+                            <Grid item xs={12}>
+                                <ConfigSwitchInput
+                                    value={this.state.failedDependencies}
+                                    onValueChange={(value) => this.setState({ failedDependencies: value })}
+                                />
+                                <ConfigLabel
+                                    label="Failed Dependencies"
+                                    tooltip="Whether the bundle can depend on failed/killed
+                                    dependencies."
+                                    inline
+                                />
+                            </Grid>
+                        </Grid>
+                    </div>
+                )}
+            >
+                {/* Main Content ------------------------------------------------------- */}

```

Comment:
would be great if we can break these huge blocks into smaller functions representing logical pieces of the interface.

Reply:
See above.